src/main/java/io/github/nucleuspowered/nucleus/modules/kit/commands/firstjoin/FirstKitRedeemCommand.java

```diff
@@ -6,41 +6,64 @@
 
 import io.github.nucleuspowered.nucleus.Util;
 import io.github.nucleuspowered.nucleus.dataservices.KitService;
-import io.github.nucleuspowered.nucleus.internal.annotations.NoCooldown;
-import io.github.nucleuspowered.nucleus.internal.annotations.NoCost;
-import io.github.nucleuspowered.nucleus.internal.annotations.NoWarmup;
-import io.github.nucleuspowered.nucleus.internal.annotations.Permissions;
-import io.github.nucleuspowered.nucleus.internal.annotations.RegisterCommand;
+import io.github.nucleuspowered.nucleus.internal.annotations.*;
 import io.github.nucleuspowered.nucleus.internal.command.AbstractCommand;
+import io.github.nucleuspowered.nucleus.internal.command.ReturnMessageException;
 import io.github.nucleuspowered.nucleus.modules.kit.config.KitConfigAdapter;
 import org.spongepowered.api.command.CommandResult;
+import org.spongepowered.api.command.CommandSource;
 import org.spongepowered.api.command.args.CommandContext;
+import org.spongepowered.api.command.args.CommandElement;
+import org.spongepowered.api.command.args.GenericArguments;
 import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.api.text.Text;
 import org.spongepowered.api.util.Tristate;
 
 import javax.inject.Inject;
+import java.util.Optional;
 
 @SuppressWarnings("ALL")
 @Permissions(prefix = "firstjoinkit")
 @NoCost
 @NoCooldown
 @NoWarmup
 @RegisterCommand(value = "redeem", subcommandOf = FirstKitCommand.class)
-public class FirstKitRedeemCommand extends AbstractCommand<Player> {
-
-    @Inject
-    private KitService gds;
+public class FirstKitRedeemCommand extends AbstractCommand<CommandSource> {
+    private final String playerKey = "subject";
 
+    @Inject private KitService gds;
     @Inject private KitConfigAdapter kca;
 
     @Override
-    public CommandResult executeCommand(Player src, CommandContext args) throws Exception {
+    public CommandElement[] getArguments() {
+        return new CommandElement[] {
+                GenericArguments.onlyOne(GenericArguments.optionalWeak(GenericArguments.player(Text.of(playerKey))))
+        };
+    }
+
+    @Override
+    public CommandResult executeCommand(CommandSource src, CommandContext args) throws Exception {
+        Optional<Player> player = args.<Player>getOne(playerKey);
```

Comment:
I actually have one of two standard ways to deal with this, where the second is preferred:

1) Use `Player player = getUserFromArgs(Player.class, src, playerKey, args);` - this method will take care of the player check for you, don't need the isPresent checks etc.

https://github.com/NucleusPowered/Nucleus/blob/sponge-api/5/src/main/java/io/github/nucleuspowered/nucleus/modules/playerinfo/commands/GetPosCommand.java#L54

2) Use `AbstractCommand.SimpleTargetOtherPlayer` and implement the `executeWithPlayer(CommandSource src, Player target, CommandContext args, boolean isSelf)` - `isSelf` is true if `src` and `target` are the same.

https://github.com/NucleusPowered/Nucleus/blob/1d992f89440048d23381ba09f90fafa6bc70bef2/src/main/java/io/github/nucleuspowered/nucleus/modules/fun/commands/HatCommand.java#L37

---
This allows us to have the same console message and handling for all situations - so can you use one of these methods please?